File: COMPILADOR/kokoslan_/src/src/java/ast/KoKoPLUS.java
/**
 @author loriacarlos@gmail.com
 
*/
package src.java.ast;
import src.java.eval.KoKoContext;
import src.java.eval.KoKoNumValue;
import src.java.eval.KoKoValue;
import src.java.exception.KoKoEvalException;


public class KoKoPLUS extends KoKoBiOperation {
	
    public KoKoPLUS(KoKoAst operator, KoKoAst left, KoKoAst right){
		super(operator, left, right);
	}
	
	@Override
	public KoKoValue eval(KoKoContext ctx){
	   try {
		     KoKoNumValue lv = (KoKoNumValue)(left().eval(ctx));
			 KoKoNumValue rv = (KoKoNumValue)(right().eval(ctx));
	         return new KoKoNumValue(lv.getValue() + rv.getValue());
	   } catch (Exception e) {
			throw new KoKoEvalException(e.getMessage());
	   }
	   
   }
	
}

File: COMPILADOR/kokoslan/src/KoKoKotlin/ast/KoKoPLUS.kt
/**
 * @author loriacarlos@gmail.com
 */
package KoKoKotlin.ast

import KoKoKotlin.eval.KoKoContext
import KoKoKotlin.eval.KoKoNumValue
import KoKoKotlin.eval.KoKoValue
import KoKoKotlin.exception.KoKoEvalException
import src.java.ast.KoKoAst



class KoKoPLUS(operator: KoKoAst, left: KoKoAst, right: KoKoAst) : KoKoKotlin.ast.KoKoBiOperation(operator, left, right) {

    override fun eval(ctx: KoKoContext): KoKoValue {
        try {
            val lv = left().eval(ctx) as KoKoNumValue
            val rv = right().eval(ctx) as KoKoNumValue
            return KoKoNumValue(lv.value + rv.value)
        } catch (e: Exception) {
            throw KoKoEvalException(e.message)
        }

    }

}